katcp/resource.py

```diff
@@ -894,9 +892,7 @@ def handle_update(sensor, reading):
             # This handler is called whenever a sensor update is received
             try:
                 assert sensor is self
-                cond_matched = condition_test(reading)
-                status_matched = reading.status == status or status is None
```

Comment:
Let me explain myself better here... The `condition_or_value` parameter passed to `wait` is either a callable taking a sensor reading as parameter, or the desired sensor value itself. Line 882 quoted above simply normalises both cases to be a function for ease of use.

In the first case the parameter is already a function so becomes `condition_test` directly (and can check the status if it wants to). In the second case, we construct a function that checks the sensor value against the desired value in `condition_or_value`. Did I miss something?